[tools] 20.0 tools and SQL moved to proper directory

ridz1208 opened the pull request:
This moves 
- the `Normalize_protocol_split_rows.php` to the proper directory and updates its relative references
- the SQL output on `Normalize_Consent_Data.php` to the archive directory directly to avoid uncommitted code in the production environment's `Cleanup_patches` directory.

This also removes trialing white spaces from the end of the files

ridz1208 commented:
@zaliqarosli please give a quick run to both tools make sure they run okai (no need to test specific functionality of each)

HenriRabalais commented:
LG

zaliqarosli commented on tools/single_use/Normalise_Consent_Data.php:
```diff
@@ -307,8 +307,7 @@
     $output .= "ALTER TABLE participant_status DROP COLUMN " . $columnName . ";\n";
 }
-$filename = __DIR__ . "/../../SQL/Cleanup_patches/delete_old_consent_tables.sql";
+$filename = __DIR__ . "/../../SQL/Archive/20.0/cleanup/delete_old_consent_tables.sql";
```
`/SQL/Archive/20.0/cleanup` directory doesn't exist yet so this throws an error. should we create the cleanup directory as part of this PR?

zaliqarosli replied:
the same change should also be made to line 7 in the starting comments

ridz1208 replied:
yes

zaliqarosli commented on tools/single_use/Normalise_Consent_Data.php:
```diff
@@ -307,8 +307,7 @@
     $columnName = $column['Column_name'];
```
i am wondering if we should rename this to `NormaliZe_Consent_Data.php` to keep US/UK spelling preference consistent.

ridz1208 replied:
sure

zaliqarosli commented:
line 147 (https://github.com/aces/Loris/pull/3855/files#diff-8f297c59f9bcd85344da8f1b1cea4f5bR147) should also be updated to wherever 2018-03-01_normalise_consent.sql is currently

zaliqarosli commented:
@ridz1208 need to address requested changes above before adding "Passed Manual Tests". removing the label until issues fixed

ridz1208 commented:
@zaliqarosli I changed the destination of the files AGAIN cuz none of the other options made sense, please review the readmes and masure the script works, I also renamed it as requested

zaliqarosli commented on .gitignore:
```diff
@@ -25,4 +25,7 @@ htdocs/js/components/.module-cache/
+npm-debug.log*
+SQL/Archive/autogenerated/reversion/*.sql
+SQL/Archive/autogenerated/single_use/*.sql
+SQL/Archive/autogenerated/*.sql
```
does this need to be here if patches are not supposed to be added directly into this directory anyways? maybe it will be a good way to catch this mistake if we have it tracked by github

ridz1208 replied:
https://git-scm.com/docs/gitignore#_notes

zaliqarosli replied:
^ please elaborate

ridz1208 replied:
https://www.atlassian.com/git/tutorials/saving-changes/gitignore

zaliqarosli commented on SQL/Archive/autogenerated/single_use/README.md:
```diff
@@ -0,0 +1,5 @@
+designed to be applied on the database only once; running these scripts more 
+than once might result in an SQL failure.
+
+**Note:** Files in this directory are not tracked by github.
```
maybe add this note to the readme of the reversion directory too

zaliqarosli commented on tools/single_use/Normalize_Consent_Data.php:
```diff
@@ -307,8 +313,8 @@
 }
-$filename = __DIR__ . "/../../SQL/Cleanup_patches/delete_old_consent_tables.sql";
+
+$filename = __DIR__ . "/../../SQL/Archive/autogenerated/single_use/delete_old_consent_tables.sql";
```
change this in the intro comments on line 7 as well please

ridz1208 commented:
@zaliqarosli please review and test (this should also incorporate your check for single use of the script)

zaliqarosli commented on SQL/Archive/autogenerated/README.md:
```diff
@@ -1,5 +1,5 @@
-**Note:** SQL patches should not be generated directly into this directory; 
-they should be added to one of the subdirectories it contains
\ No newline at end of file
+**Note:** SQL patches should not be generated directly into this directory, 
```
i think a semi-colon here instead actually works better

driusan commented on .gitignore:
```diff
@@ -25,4 +25,7 @@ htdocs/js/components/.module-cache/
-npm-debug.log*
\ No newline at end of file
+npm-debug.log*
+SQL/Archive/autogenerated/reversion/*.sql
```
Why are autogenerated patches in Archive?
